prod/init-services.sh

```diff
@@ -0,0 +1,239 @@
+  )
+
+  for file in "${files[@]}"; do
+    if [[ -f "$file" ]]; then
+      local output="${file%.example}"  # Remove .example extension
+      envsubst < "$file" > "$output" && echo "Generated: $output"
+    else
+      echo "Warning: Template file $file not found."
+    fi
+  done
+}
+
+# Replace variables directly in .env files
+update_env_files() {
+  local files=(
+    './services/wiki/php.env.example'
+    './services/wiki/mariadb.env.example'
+    './services/online/php.env.example'
+    './services/online/mariadb.env.example'
+    './services/click/php.env.example'
+    './services/click/mariadb.env.example'
+    './services/danger/php.env.example'
+    './services/danger/mariadb.env.example'
+  )
+
+  for file in "${files[@]}"; do
+    if [[ -f "$file" ]]; then
+      echo "Updating variables in: $file"
+
+      # Create a backup before making any changes
+      cp "$file" "${file}.bak"
+
+      # Substitute service-specific secrets
+      for item in "${SERVICE_ITEMS[@]}"; do
+        local service_name="${item//-/_}"
+        local root_password_var="${service_name^^}_MARIADB_ROOT_PASSWORD"
+        local password_var="${service_name^^}_MARIADB_PASSWORD"
+
+        # Get the values for passwords from the environment variables
+        local mariadb_root_password="${!root_password_var}"
+        local mariadb_password="${!password_var}"
+
+        # Perform in-place substitution
+        sed -i "s|${service_name^^}_MARIADB_ROOT_PASSWORD=.*|${service_name^^}_MARIADB_ROOT_PASSWORD=$mariadb_root_password|" "$file"
+        sed -i "s|${service_name^^}_MARIADB_PASSWORD=.*|${service_name^^}_MARIADB_PASSWORD=$mariadb_password|" "$file"
+      done
+
+      # Substitute the global secrets
+      sed -i "s|DB_PASSWORD=.*|DB_PASSWORD=$DB_PASSWORD_WIKI|" "$file"
+      sed -i "s|LOGIN_PASSWORD=.*|LOGIN_PASSWORD=$LOGIN_PASSWORD_WIKI|" "$file"
+      sed -i "s|ADMIN_PASSWORD=.*|ADMIN_PASSWORD=$ADMIN_PASSWORD_WIKI|" "$file"
+      sed -i "s|BAN_PASSWORD=.*|BAN_PASSWORD=$BAN_PASSWORD_WIKI|" "$file"
+      sed -i "s|SITE_URL=.*|SITE_URL=$SITE_URL|" "$file"
+      sed -i "s|ALLOWED_EMBEDS=.*|ALLOWED_EMBEDS=$ALLOWED_EMBEDS|" "$file"
+
+      echo "Successfully updated $file"
+    else
+      echo "Warning: .env file $file not found."
+    fi
+  done
+}
+
+# Configure Traefik environment file
+config_traefik() {
+  cp ./traefik/traefik.env.example ./traefik/traefik.env
+}
+
+# Change to the script's directory
+set_script_dir() {
+  local dirname
+  dirname=$(dirname "$0")
+  SCRIPT_DIR=$(cd "$dirname" || exit; pwd)
+  cd "$SCRIPT_DIR" || exit
+}
+
+# Check Docker Compose version
+check_docker_compose() {
+  if ! docker compose version &>/dev/null; then
+    echo "Error: Docker Compose is not installed or not found."
+    exit 2
+  fi
+}
+
+# Run Docker Compose commands
+run_docker_compose() {
+  local action="$1"
+  local project_dirs=("traefik" "services/home" "services/online" "services/wiki" "services/click" "services/danger")
```

Comment:
Glitch is also missing here. Same question as above. Why is this hardcoded instead of looking at all of the submodules in the directory?